src/Compilers/CSharp/Portable/Symbols/FieldSymbol.cs

```diff
@@ -453,5 +453,20 @@ protected override ISymbol CreateISymbol()
         {
             return new PublicModel.FieldSymbol(this);
         }
+
+        public override bool Equals(Symbol other, TypeCompareKind compareKind)
+        {
+            if (other is SubstitutedFieldSymbol sfs)
+            {
+                return sfs.Equals(this, compareKind);
```

Comment:
Because the substituted ones have special logic to compare against the non-substited. We've been through various iterations of solving this and this is sadly the simplest / easiest to reason about.